### ISSUE TITLE
allow variable as array index

### ISSUE THREAD

**Issue by mmrj** [forio/flow.js#9]
```data-f-bind="sales[<sampleInt>]```  (or some alternate syntax?) should work at least when sampleInt is a valid array index for sales.

**Comment by mmrj** [comment]
(re)confirmed this is verified in 0.8
reopening because this is broken again in 0.8.3

**Comment by narenranjit** [comment]
@mmrj I have a testpage (with your model) with

```
 <td>span: <span data-f-bind="sales[<sampleTruthy>]"></span><br>input: <input type="text" data-f-bind="sales[<sampleTruthy>]" />
        <br> check: sampleTruthy is <span data-f-bind="sampleTruthy"></span> and sales[1] is <span data-f-bind="sales[1]"></span>
    </td>
```
I see the values populated correctly. What's a case which doesn't work? (point me to a line in sandbox-mj if that's where this is)

**Comment by mmrj** [comment]
if i use the input element to try and update, i get an error (400) on the PATCH,  -- although that's the opposite behavior of the comment i left myself for that test, so i'm a little stumped and wondering if this is just a fragile area, and/or if "0.8.3" has changed since i did this testing.

at any rate, this is line number 87-106 in https://github.com/forio/flow.js/blob/tests-0.9.0/tests/integration/sandbox-mj.html (note the branch). maybe let's review together when we meet next.

**Comment by narenranjit** [comment]
0.8.3 hasn't changed - my changes are in a different branch (feature/auto-fetch-variables), though there weren't any changes to address this specifically. Will try it there

**Comment by mmrj** [comment]
this is fixed in 0.9.0. verified.
